src/libcore/panicking.rs

```diff
@@ -52,6 +52,24 @@ pub fn panic(expr: &str) -> ! {
     panic_fmt(fmt::Arguments::new_v1(&[expr], &[]), Location::caller())
 }
 
+#[cfg(not(bootstrap))]
+#[cold]
+#[cfg_attr(not(feature = "panic_immediate_abort"), inline(never))]
+#[track_caller]
+#[lang = "panic_bounds_check"] // needed by codegen for panic on OOB array/slice access
+fn panic_bounds_check(index: usize, len: usize) -> ! {
+    if cfg!(feature = "panic_immediate_abort") {
+        unsafe { super::intrinsics::abort() }
+    }
+
+    panic_fmt(
```

Comment:
Any chance we can change `panic_fmt` too? I forget if @anp tried it.

Reply:
I think there might be a problem doing this for the bootstrap build; what should the `cfg(bootstrap)` version of `fn panic` do with its `location` if it cannot forward it to `panic_fmt`?

But I guess this could be changed after a bootstrap bump.

Reply:
Oh does `fn panic`also have a location argument? Is that actually used in practice?

Reply:
Sorry I meant the `cfg(bootstrap)` version of `fn panic_bounds_check` -- the old version that I had to leave around here to match the bootstrap compiler.